options/rtdl/generic/linker.cpp

```diff
@@ -108,14 +111,53 @@ SharedObject *ObjectRepository::injectStaticObject(frg::string_view name,
 	return object;
 }
 
+namespace {
+frg::string<MemoryAllocator> normalizePath(frg::string_view name) {
+	frg::vector<frg::string<MemoryAllocator>, MemoryAllocator> stack {
+		getAllocator()
+	};
+	size_t curr;
+	size_t prev = 0;
+	auto next = [&] () {
+		curr = name.find_first('/', prev);
+	};
+	auto doComponent = [&] (frg::string_view component) {
+		if (!component.size()) {
+			return;
+		}
+		if (component == ".") {
+			return;
+		}
+		if (component == "..") {
+			stack.pop();
```

Comment:
This is broken for symlinks.

Reply:
the actual breakage to fix is in managarm-system, so this code could be removed

Reply:
Agreed. Path normalization should not be done here.